internal/provisioner/handler.go

```diff
@@ -128,9 +137,9 @@ func (h *eventHandler) HandleEventBatch(ctx context.Context, batch events.EventB
 	h.ensureDeploymentsMatchGateways(ctx)
 }
 
-func generateDeploymentID(gatewayNsName types.NamespacedName) string {
-	// for production, make sure the ID is:
-	// - a valid resource name (ex. can't be too long);
-	// - unique among all Gateway resources (Gateways test-test/test and test/test-test should not have the same ID)
-	return fmt.Sprintf("nginx-gateway-%s-%s", gatewayNsName.Namespace, gatewayNsName.Name)
+func (h *eventHandler) generateDeploymentID() string {
```

Comment:
Could we generate a random UUID instead? Or is that too unpredictable?

Reply:
I don't think it make sense at this point:
- it will complicate testing
- if the provisioner restarts, it will generate new deployments. (right now the provisioner will fail, because it will try to create deployments that already exists, which is ok for the purposes of running conformance tests).